src/ethereum/networkAdapters/PulsechainScanAdapter.ts

```diff
@@ -0,0 +1,262 @@
+import { mul, sub } from 'biggystring'
+import {
+  asArray,
+  asBoolean,
+  asJSON,
+  asNumber,
+  asObject,
+  asOptional,
+  asString
+} from 'cleaners'
+import { EdgeTransaction } from 'edge-core-js/types'
+
+import { EthereumNetworkUpdate, getFeeRateUsed } from '../EthereumNetwork'
+import { EthereumTxOtherParams } from '../ethereumTypes'
+import { GetTxsParams, NetworkAdapter } from './types'
+
+export interface PulsechainScanAdapterConfig {
+  type: 'pulsechain-scan'
+  servers: string[]
+}
+
+export class PulsechainScanAdapter extends NetworkAdapter<PulsechainScanAdapterConfig> {
+  fetchBlockheight = null
+  broadcast = null
+  getBaseFeePerGas = null
+  multicastRpc = null
+  fetchNonce = null
+  fetchTokenBalance = null
+  fetchTokenBalances = null
+
+  currentScan: Promise<EthereumNetworkUpdate> | undefined
+
+  fetchTxs = async (params: GetTxsParams): Promise<EthereumNetworkUpdate> => {
+    try {
+      // We shouldn't start scanning if scanning is already happening:
+      if (this.currentScan != null) {
+        return await this.currentScan
+      }
+
+      this.currentScan = this.checkTransactions(params)
+      const update = await this.currentScan
+      return update
+    } catch (error) {
+      console.error(error)
+      throw error
+    } finally {
+      this.currentScan = undefined
+    }
+  }
+
+  private async checkTransactions(
+    params: GetTxsParams
+  ): Promise<EthereumNetworkUpdate> {
+    const { startBlock, currencyCode } = params
+    const address = this.ethEngine.walletLocalData.publicKey
+
+    const scanTransactions: PulsechainScanTransaction[] = []
+    let nextPageParams: NextPageParams | undefined
+
+    while (true) {
+      const responseData = await this.queryTransactions(address, nextPageParams)
+
+      scanTransactions.push(...responseData.items)
+      nextPageParams = responseData.next_page_params
+
+      // Exit if a transaction block height is less than/equal to the startBlock:
+      if (responseData.items.some(scanTx => scanTx.block <= startBlock)) break
+
+      // Exit if there is no next page of transactions:
+      if (nextPageParams == null) break
+    }
+
+    // Convert the transaction data into EdgeTransactions:
+
+    const edgeTransactions: EdgeTransaction[] = scanTransactions.map(tx =>
+      this.processScanTransaction(tx, currencyCode)
+    )
+
+    return {
+      tokenTxs: {
+        [currencyCode]: {
+          blockHeight: startBlock,
+          edgeTransactions
+        }
+      },
+      server: this.config.servers.join(',')
+    }
+  }
+
+  private processScanTransaction(
```

Comment:
This method could be a standalone function, instead of a class member. It never touches `this`

Reply:
It uses `this` four times.